Tormented Demon Plugin

changednamedes opened the pull request:
**Features**

- **Full Automation:** Automatically handles banking, traveling, fighting, and retreating.
    -   **Banking & Restocking:** Uses a pre-defined inventory setup named "tormented" for withdrawing/restocking supplies. Ensure a duelling ring is included for teleporting.
    -   **Traveling & Combat:** Automatically travels to Tormented Demons and engages in combat. **(MUST have Guthixian temple teleport)**
    -   **Auto-Retreat:** Returns to safety if supplies fall below safe levels, restocks, and resumes **(MUST have duelling ring as it uses Ferox Enclave )**

- **Combat Only Mode:** Starts directly in combat mode, focusing only on fighting without restocking or traveling.
    - **Setup:** Position your character near the Tormented Demons, and the bot will take care of the combat, health, and prayer management **(This does NOT retreat)**

- **Auto Prayer & Gear Switching:** Adjusts prayers and gear based on the demon's attack style to optimize defence and offense.
- **Loot Management:** Configure specific items to loot by name.

**Starting the Plugin**

**1. Configure Settings:** Adjust settings as needed (Full Auto or Combat Only, looting preferences, prayer switching, etc.)
**2. Start the Plugin:**
    - Full Auto: Ensure the "tormented" inventory setup exists and contains all needed supplies, including a duelling ring.
    - Combat Only: Position your character near the demons and let the bot handle the combat.

changednamedes committed:
Tormented

changednamedes committed:
Merge remote-tracking branch 'origin/development' into tormenteddemons

changednamedes committed:
v1.0.0

changednamedes committed:
v1.0.0

changednamedes committed:
v1.0.0

changednamedes committed:
v1.0.0

changednamedes committed:
changed it so it doesn't deposit equiptment and also fixed the looting when only one inventory slot, messes up when changing gear

changednamedes committed:
Changed delay as didn't need to be so high and sometimes misses the dodge with mage gear

Commit:
Fixed travelStep not updating. Now it won't get stuck restocking.

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonPlugin.java:
```diff
@@ -0,0 +1,138 @@
+@PluginDescriptor(
+        name = PluginDescriptor.zerozero + "Tormented Demons",
+        description = "Automates restocking, prayer flicking, and gear switching during Tormented Demon",
+        tags = {"tormented", "flicker", "weapon", "switch"},
```
Can you add "microbot" as a tag? it's easier to filter if people are searching for microbot scripts

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonPlugin.java:
```diff
@@ -0,0 +1,138 @@
+
+    @Subscribe
+    private void onConfigChanged(ConfigChanged event) {
+        if (event.getGroup().equals("tormenteddemon")) {
```
Try to work with early returns to avoid indenting.

```
if (!event.getgroup().equals("tormenteddemon")) {
    return;
}
```

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonPlugin.java:
```diff
@@ -0,0 +1,138 @@
+        final GraphicsObject graphicsObject = event.getGraphicsObject();
+        final int TORMENTED_VENGENCE_SPECIAL = 2856;
+
+        if (graphicsObject.getId() == TORMENTED_VENGENCE_SPECIAL) {
```
Same here, early return

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonPlugin.java:
```diff
@@ -0,0 +1,138 @@
+            } else if (event.getKey().equals("combatOnly") && config.combatOnly()) {
+                configManager.setConfiguration("tormenteddemon", "fullAuto", false);
+            }
+            if (event.getKey().equals("copyGear")) {
```
Interesting...what is the copy gear used for?

changednamedes replied:
This copies the names of your current gear to a comma separated string, so you can add them to the gear switcher quicker/easier

chsami replied:
oh cool, def something we could use for other plugins aswell

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonScript.java:
```diff
@@ -0,0 +1,501 @@
+    private BankingStep bankingStep = BankingStep.DRINK;
+
+    public boolean run(TormentedDemonConfig config) {
+        if (config.fullAuto()) {
```
Since only one of the two options is possible I think it's more logical to have an enum in your config with the available options. This will then display as a dropdown.

Example in tithe farm script: 

![image](https://github.com/user-attachments/assets/1da6f623-fb42-4592-ba3d-61006ad581e3)

changednamedes replied:
All done

changednamedes committed:
Updated for review.

Added early returns.
Added Enum for mode.
Added tag.

changednamedes committed:
Added teleportToFeroxEnclave()
- This way we can interact with the ring instead of trying to go through the teleport menu with the walker, for the retreating. It won't be interrupted now.

changednamedes committed:
Fixing NPC being null after restock
